src/http/httpProxy.ts

```diff
@@ -117,3 +136,37 @@ export async function setupHttpProxy(): Promise<string> {
         });
     });
 }
+
+// Function to get a random port within a specified range
+function getRandomPort(): number {
+    return Math.floor(Math.random() * (maxPort - minPort + 1)) + minPort;
+}
+
+// Function to find an open port starting from a specified port
+function findOpenPort(callback: (port: number) => void): void {
+    const server = net.createServer();
+
+    function tryPort(port: number) {
+        server.once('error', () => {
```

Comment:
This will retry finding a new port for all errors. How about we check if (err.code === 'EADDRINUSE')  and then find the next port.